llvm/lib/Target/AArch64/AArch64SchedNeoverseV2.td

```diff
@@ -1108,12 +1108,18 @@ def : InstRW<[V2Write_1cyc_1B_1R], (instrs BL, BLR)>;
 // ALU, basic
 // ALU, basic, flagset
 def : SchedAlias<WriteI, V2Write_1cyc_1I>;
-def : InstRW<[V2Write_1cyc_1F], (instregex "^(ADC|SBC)S[WX]r$")>;
+def : InstRW<[V2Write_1cyc_1F], (instregex "^(ADD|SUB)S[WX]r[ir]$",
```

Comment:
I believe `r` is wrong in the last group, should be `s`, but that's already captured a few lines down.
```suggestion
def : InstRW<[V2Write_1cyc_1F], (instregex "^(ADD|SUB)S[WX]ri$",
```
Also, it is not clear to me if for ADD{S}/SUB{S} (immediate), this should only apply for "LSL shift, shift <= 4". Have you tested with `LSL #12`?

Reply:
ADDSWrr is a psuedo that is used during codegen and turned into a ADDSWrs with a zero shift. I believe it should be treated like a (cheap) ADDSWrs for scheduling.

And for ri I believe they are constants either way, so does not apply to the same "free vs expensive shifts" as in the ADDrs variants. i.e they are always "ALU, basic, flagset".

Reply:
I see, perfect, thanks for clarifying.